framework/Models/EnsembleModel.py

```diff
@@ -707,13 +686,93 @@ def _externalRun(self,inRun, jobHandler):
         residualPass = residueContainer['TotalResidue'] <= self.convergenceTol
         # sometimes there can be multiple residual values
         if hasattr(residualPass,'__len__'):
-          residual = all(residualPass)
+          residualPass = all(residualPass)
         if residualPass:
           self.raiseAMessage("Picard's Iteration converged. Norm: "+ str(residueContainer['TotalResidue']))
           break
     returnEvaluation = returnDict, inRunTargetEvaluations, tempOutputs
     return returnEvaluation
 
+  def __advanceModel(self, identifier, modelToExecute, origInputList, inputKwargs, inRunTargetEvaluations, samplerType, iterationCount, jobHandler = None):
+    """
+      This method is aimed to advance the execution of a sub-model and to collect the data using
+      the realization
+      @ In, identifier, str, current job identifier
+      @ In, modelToExecute, super(Model), Model instance than needs to be avanced
+      @ In, origInputList, list, list of model input
+      @ In, inputKwargs, dict, dictionary of kwargs for this model
+      @ In, inRunTargetEvaluations, DataObject, target evaluation for the model to advance
+      @ In, samplerType, str, sampler Type
+      @ In, iterationCount, int, iteration counter (1 if not picard)
+      @ In, jobHandler, jobHandler instance, optional, jobHandler instance (available only if parallelStrategy == 2)
+      @ Out, returnDict, dict, dictionary containing the data extracted from the target evaluation
+      @ Out, gotOutputs, dict, dictionary containing all the data coming out the model
+      @ Out, evaluation, dict, the evaluation dictinary with the "unprojected" data
+    """
+    returnDict = {}
+
+    self.raiseADebug('Submitting model',modelToExecute['Instance'].name)
+    localIdentifier =  modelToExecute['Instance'].name+utils.returnIdSeparator()+identifier
+    if self.parallelStrategy == 1:
+      # we evaluate the model directly
+      try:
+        evaluation = modelToExecute['Instance'].evaluateSample.original_function(modelToExecute['Instance'], origInputList, samplerType, inputKwargs)
+      except Exception:
+        evaluation = None
+    else:
+      nextModel = False
```

Comment:
I think (unless I am misreading the code), nextModel and the ```while not nextModel``` loop could be removed and this would simplify the code.

Reply:
you are right...it was a left over of an old implementation...fixed